### PR TITLE
Remove CachedDfSource

### PR DESCRIPTION
**Pull request by martijnvg**
and move the dfs logic to the ContextIndexSearcher.

This PR relates to #12864

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java [hunk 1]
```diff
@@ -77,8 +77,8 @@ public Query rewrite(Query original) throws IOException {
     public Weight createNormalizedWeight(Query query, boolean needsScores) throws IOException {
         try {
             // if scores are needed and we have dfs data then use it
-            if (dfSource != null && needsScores) {
-                return dfSource.createNormalizedWeight(query, needsScores);
+            if (aggregatedDfs != null && needsScores) {
+                return super.createNormalizedWeight(query, needsScores);
             }
             return in.createNormalizedWeight(query, needsScores);
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java]: Wouldn't the 3 above lines be equivalent to just doing: `return super.createNormalizedWeight(query, needsScores);`?

**Comment by jpountz** [comment]
LGTM, just left a minor suggestion

File: core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java (hunk 1 above)
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java]: During tests the wrapped `IndexSearcher` is an `AssertingIndexSearch` and if we wouldn't delegate to `in` we would miss assertions during tests?

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java]: ok, can you add a comment saying this?

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java]: yes, I can :)

On 19 August 2015 at 10:23, Adrien Grand <notifications@github.com> wrote:

> In
> core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java
> <https://github.com/elastic/elasticsearch/pull/12973#discussion_r37390259>
> :
>
> > @@ -77,8 +77,8 @@ public Query rewrite(Query original) throws IOException {
> >      public Weight createNormalizedWeight(Query query, boolean needsScores) throws IOException {
> >          try {
> >              // if scores are needed and we have dfs data then use it
> > -            if (dfSource != null && needsScores) {
> > -                return dfSource.createNormalizedWeight(query, needsScores);
> > +            if (aggregatedDfs != null && needsScores) {
> > +                return super.createNormalizedWeight(query, needsScores);
> >              }
> >              return in.createNormalizedWeight(query, needsScores);
>
> ok, can you add a comment saying this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/12973/files#r37390259>.
>



-- 
Met vriendelijke groet,

Martijn van Groningen

**Pull request merged**